src/org/omegat/core/machinetranslators/MTHUBTranslate.java

```diff
@@ -0,0 +1,481 @@
+    }
+}
+
+/**
+ * A class for keep the language response from MT-Hub.
+*/
+class MTHUBLanguageResponse {
+    public boolean success;
+    public List<MTHUBResponseError> error;
+    public List<MTHUBLanguageResponseData> data;
+
+    public boolean isSuccess() {
+        return success;
+    }
+
+    public void setSuccess(boolean success) {
+        this.success = success;
+    }
+
+    public List<MTHUBResponseError> getError() {
+        return error;
+    }
+
+    public void setError(List<MTHUBResponseError> error) {
+        this.error = error;
+    }
+
+    public List<MTHUBLanguageResponseData> getData() {
+        return data;
+    }
+
+    public void setData(List<MTHUBLanguageResponseData> data) {
+        this.data = data;
+    }
+}
+/**
+ * MT-HUB (https://mt-hub.eu/)
+ * @author prompsit (help@prompsit.com)
+ * @see <a href="https://iadaatpa.docs.apiary.io/">Translation API</a>
+ */
+public class MTHUBTranslate extends BaseTranslate {
+    private static final String PROPERTY_API_KEY = "mthub.api.key";
+    protected static final String GT_URL = "https://app.mt-hub.eu/api/translate";
+    protected static final int LIMIT_CHARACTER = 2000;
+    private List<String> availableLanguageCodes = null;
+    private static final ObjectMapper OM = new ObjectMapper();
+    static {
+        OM.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY,
+                        true);
+    }
+
+    /**
+     * Get from mt-hub the available language codes and set into the arraylist.
+    */
+    private List<String> fetchAvailableLanguageCodes() {
+
+       String mthubKey = getCredential(PROPERTY_API_KEY);
+       String encoding = StandardCharsets.UTF_8.name();
+
+       try {
+            List<String> codes = new ArrayList<String>();
+            URL url = new URL("https://app.mt-hub.eu/api/describelanguages/"
+                              + mthubKey);
+            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+
+            conn.setDoOutput(true);
+            conn.setRequestProperty("Accept-Charset", encoding);
+            conn.setRequestProperty("Content-Type",
+                                        "application/json;charset=" + encoding);
+            int statusCode = conn.getResponseCode();
+
+            try (InputStream in = (statusCode >= 200 && statusCode < 400)
+                    ? conn.getInputStream() : conn.getErrorStream();) {
+                MTHUBLanguageResponse response = OM.readValue(IOUtils.toString(in, StandardCharsets.UTF_8),
+                                                      MTHUBLanguageResponse.class);
+                if (response.isSuccess()) {
+                    for (List<MTHUBLanguageResponseDataLanguages> dl : response.getData().get(0).getLanguages()) {
+                        codes.add(dl.get(0).getCode().toUpperCase());
+                    }
+                    return codes;
+                } else {
+                    System.err.println("IOException: "
```

Comment:
Instead of printing to stderr these should be logged with a `java.util.logging.Logger`.

Reply:
add LOGGER.log